library/std/src/sys/wasi/mod.rs

```diff
@@ -82,31 +82,99 @@ pub fn is_interrupted(errno: i32) -> bool {
 }
 
 pub fn decode_error_kind(errno: i32) -> std_io::ErrorKind {
-    use std_io::ErrorKind::*;
-    if errno > u16::MAX as i32 || errno < 0 {
-        return Uncategorized;
+    use std_io::ErrorKind;
+
+    let Ok(errno) = u16::try_from(errno) else {
+        return ErrorKind::Uncategorized;
+    };
+
+    macro_rules! match_errno {
+        ($($($errno:ident)|+ => $errkind:ident),*, _ => $wildcard:ident $(,)?) => {
+            match errno {
+                $(e if $(e == ::wasi::$errno.raw())||+ => ErrorKind::$errkind),*,
+                _ => ErrorKind::$wildcard,
+            }
+        };
     }
 
-    match errno {
-        e if e == wasi::ERRNO_CONNREFUSED.raw().into() => ConnectionRefused,
-        e if e == wasi::ERRNO_CONNRESET.raw().into() => ConnectionReset,
-        e if e == wasi::ERRNO_PERM.raw().into() || e == wasi::ERRNO_ACCES.raw().into() => {
-            PermissionDenied
-        }
-        e if e == wasi::ERRNO_PIPE.raw().into() => BrokenPipe,
-        e if e == wasi::ERRNO_NOTCONN.raw().into() => NotConnected,
-        e if e == wasi::ERRNO_CONNABORTED.raw().into() => ConnectionAborted,
-        e if e == wasi::ERRNO_ADDRNOTAVAIL.raw().into() => AddrNotAvailable,
-        e if e == wasi::ERRNO_ADDRINUSE.raw().into() => AddrInUse,
-        e if e == wasi::ERRNO_NOENT.raw().into() => NotFound,
-        e if e == wasi::ERRNO_INTR.raw().into() => Interrupted,
-        e if e == wasi::ERRNO_INVAL.raw().into() => InvalidInput,
-        e if e == wasi::ERRNO_TIMEDOUT.raw().into() => TimedOut,
-        e if e == wasi::ERRNO_EXIST.raw().into() => AlreadyExists,
-        e if e == wasi::ERRNO_AGAIN.raw().into() => WouldBlock,
-        e if e == wasi::ERRNO_NOSYS.raw().into() => Unsupported,
-        e if e == wasi::ERRNO_NOMEM.raw().into() => OutOfMemory,
-        _ => Uncategorized,
+    match_errno! {
+        ERRNO_2BIG           => ArgumentListTooLong,
```

Comment:
Can't strip the `ERRNO_` prefix as `2BIG` starts with a digit and doesn't match in `ident`.

Reply:
That's fine. Correctly naming the constant is acceptable, repeating everything else was not.